subsys/bluetooth/mesh/transport.c

```diff
@@ -1019,6 +1013,106 @@ int bt_mesh_ctl_send(struct bt_mesh_net_tx *tx, u8_t ctl_op, void *data,
 	return bt_mesh_net_send(tx, buf, cb, cb_data);
 }
 
+static inline int ctl_send_seg(struct bt_mesh_net_tx *tx, u8_t ctl_op, void *data,
+				     size_t data_len, u64_t *seq_auth,
+				     const struct bt_mesh_send_cb *cb, void *cb_data)
+{
+	struct net_buf *buf;
+	u8_t seg_o = 0;
+	u16_t seq_zero;
+	struct seg_tx *tx_seg;
+	u8_t segment_count = (data_len - 1) / 8; /* Number of segments-1 */
+	int i;
+
+	for (tx_seg = NULL, i = 0; i < ARRAY_SIZE(seg_tx); i++) {
+		if (!seg_tx[i].nack_count) {
+			tx_seg = &seg_tx[i];
+			break;
+		}
+	}
+
+	if (!tx_seg) {
+		BT_ERR("No multi-segment message contexts available");
+		return -EBUSY;
+	}
+
+	tx_seg->dst = tx->ctx->addr;
+	tx_seg->seg_n = segment_count;
+	tx_seg->nack_count = tx_seg->seg_n + 1;
+	tx_seg->seq_auth = SEQ_AUTH(BT_MESH_NET_IVI_TX, bt_mesh.seq);
+	tx_seg->sub = tx->sub;
+	tx_seg->new_key = tx->sub->kr_flag;
+	tx_seg->cb = cb;
+	tx_seg->cb_data = cb_data;
+
+	if (tx->ctx->send_ttl == BT_MESH_TTL_DEFAULT) {
+		tx_seg->ttl = bt_mesh_default_ttl_get();
+	} else {
+		tx_seg->ttl = tx->ctx->send_ttl;
+	}
+
+	seq_zero = tx_seg->seq_auth & 0x1fff; /* least 13 bits of SeqAuth */
```

Comment:
Minor (and I think you just copied the same from the access message code): you could use `BIT_MASK(13)`